key_test.go

```diff
@@ -0,0 +1,68 @@
+package shell
+
+import (
+	"context"
+	"testing"
+
+	"github.com/cheekybits/is"
+)
+
+func TestKeyGen(t *testing.T) {
+	is := is.New(t)
+	s := NewShell(shellUrl)
+
+	key, err := s.KeyGen(context.Background(), "testKey", KeyGen.Type("ed25519"), KeyGen.Size(2048))
+	is.Nil(err)
```

Comment:
While this passes when run against go-ipfs it's essentially a bug. ed25519 implicitly has a size associated with it (and that size is 256). Instead we should test:

1. ed25519 with no size works
2. rsa with no size works
3. rsa with a valid custom size (e.g. 4096) works
4. rsa with an invalid custom size (e.g. 1024, which is too small) fails